### ISSUE TITLE
Support an executable based recommendation by executable prefix name

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#235286]
MongoDB compass does not seem to be recommended for me on my machine.
Code pointer https://github.com/microsoft/vscode-distro/blob/main/mixin/insider/product.json#L875

MacOS

1. Install https://www.mongodb.com/products/tools/compass
2. Open VS Code Insiders
3. No recommendation 🐛

**Comment by GaurabAryal** [comment]
@sandy081 – looping in @alenakhineika from MongoDB engineering team if you needed help with debugging / bouncing off ideas. We could also explore other ways to recommend MongoDB if this path doesn't end up working out.

**Comment by GaurabAryal** [comment]
An additional comment, if helpful: I tested on Windows as well with Compass in `C:\Program Files\MongoDB Compass`. Still did not see the recommendation show up (screenshot below).


![Image](https://github.com/user-attachments/assets/a122700b-8c76-4532-9cb9-94b4eb3f6dcd)

**Comment by sandy081** [comment]
@isidorn I do not see any `mongo compass` executable installed in the path `/usr/local/bin/`. I am only seeing following

<img width="712" alt="Image" src="https://github.com/user-attachments/assets/11f9100c-88f5-4221-b52a-6a3161cad9d1" />

Hence it is not getting recommended. Can you please confirm if this is the right executable name?

**Comment by isidorn** [comment]
Thanks @sandy081, I thought that you would look into the Applications folder.

@GaurabAryal @alenakhineika can you please check which of these executables in `/usr/local/bin/` we should be detecting for?

**Comment by GaurabAryal** [comment]
@isidorn @sandy081 – actually all of those executables are relevant in this situation, i.e. they point to the developer being a MongoDB customer. Is it possible to glob this to `/usr/local/bin/mongo*`? This would also cover the core MongoDB Shell which isn't listed there (`mongosh`). Here's my `/usr/local/bin/mongosh` (Mac).
<img width="513" alt="Image" src="https://github.com/user-attachments/assets/d35c3ff2-1c82-43e4-b70f-e7fdef020d45" />

Also to clarify, are you looking in a linux box currently? For Compass it might be `/usr/local/bin/compass` - could you try that out?

Finally, for Mac & Windows, a good path for Compass would definitely be the Applications folder & the Program Files folder, respectively. It is a little hard to tell as all of this depends on a) if the user installed in a non-default directory and b) the OS. Appreciate you working with us on resolving the issue :-).

**Comment by sandy081** [comment]
> Is it possible to glob this to /usr/local/bin/mongo*

Yeah, this can be supported. Just wondering if there is some mongo executable that will exists always? If so we can just point to that.

**Comment by GaurabAryal** [comment]
> Yeah, this can be supported. Just wondering if there is some mongo executable that will exists always? If so we can just point to that.

Not exactly since we have a handful of tools that could be installed, serving various purposes (i.e. MongoDB Compass is a GUI data browser while MongoDB Shell is a REPL data browser).
